controllers/platformnetwork_controller.go

```diff
@@ -661,38 +741,261 @@ func (r *PlatformNetworkReconciler) statusUpdateRequired(instance *starlingxv1.P
+				}
+
+				if default_system.SystemMode == "simplex" && strings.ToLower(default_system.SystemType) == "all-in-one" {
+					return &host, true, nil
+				}
 			}
 		}
+	}
 
-		inSync := err == nil
+	return nil, false, nil
+}
 
-		if inSync != instance.Status.InSync {
-			r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "synchronization has changed to: %t", inSync)
+func (r *PlatformNetworkReconciler) AddressPoolUpdateRequired(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork) (bool, error) {
+	var pool *addresspools.AddressPool
+	var err error
+	pool, err = r.FindExistingAddressPool(client, instance)
+	if err != nil {
+		return false, err
+	} else if pool == nil {
+		return true, nil
+	}
+
+	if _, ok := poolUpdateRequired(instance, pool, r); ok {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (r *PlatformNetworkReconciler) NetworkUpdateRequired(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork) (bool, error) {
+	if instance.Spec.Type == cloudManager.OAMNetworkType {
+		oamNetworkList, err := oamNetworks.ListNetworks(client)
+		if err != nil {
+			return false, err
+		} else if len(oamNetworkList) == 0 {
+			return true, nil
+		} else {
+			oamNetwork := &oamNetworkList[0]
+			if _, ok := oamUpdateRequired(instance, oamNetwork, r); ok {
+				return true, nil
+			}
+		}
+	} else {
+		network, err := r.FindExistingNetwork(client, instance)
+		if err != nil {
+			return false, err
+		} else if network == nil {
+			return true, nil
 		}
+		if _, ok := plNetworkUpdateRequired(instance, network); ok {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func (r *PlatformNetworkReconciler) PlatformNetworkUpdateRequired(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork) (bool, error) {
+	pool_update_required, err := r.AddressPoolUpdateRequired(client, instance)
+	if err != nil {
+		err = common.NewSystemDependency("There was an error retrieving the address pool.")
+		return false, err
+	}
 
+	nwk_update_required, err := r.NetworkUpdateRequired(client, instance)
+	if err != nil {
+		err = common.NewSystemDependency("There was an error retrieving the network.")
+		return false, err
+	}
+
+	return pool_update_required || nwk_update_required, err
+}
+
+func (r *PlatformNetworkReconciler) UpdateInsyncStatus(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork, err error) (bool, error) {
+	inSync := err == nil
+	if inSync != instance.Status.InSync || inSync != instance.Status.Reconciled {
+		r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "synchronization has changed to: %t", inSync)
 		instance.Status.InSync = inSync
+		instance.Status.Reconciled = inSync
+		err2 := r.Client.Status().Update(context.TODO(), instance)
+		if err2 != nil {
+			logPlatformNetwork.Error(err2, "failed to update platform network status")
+			return inSync, err2
+		}
+	}
+	return inSync, nil
+}
 
-		if !instance.Status.Reconciled {
```

Comment:
where is this code getting removed from ? either of the above and below functions..

Reply:
Suggest to compare with original file for clarity, these diffs might be misleading due to too many changes.